plinky_elf/src/reader/sections/mod.rs

```diff
@@ -124,11 +124,14 @@ fn read_section_inner(
         }
     };
 
+    let is_retain = header.flags.gnu_retain;
```

Comment:
This flag is inside the `SHF_MASKOS` range, so it's specific to `ELFOSABI_GNU`. Could you add a `// TODO: limit it only to ELFOSABI_GNU`?